### PR TITLE
Add 'add_row_if' and 'add_rows_if' methods

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds the ability to only add (a) row(s) when the specified predicate evaluates to `true`. This enables a smoother "flow" when building tables. For example:

```rust
let mut table = Table::new();
table
    .set_header(&vec!["Header1", "Header2", "Header3"])
    .add_row(&vec![
        "This is a text",
        "This is another text",
        "This is the third text",
    ]);
if predicate {
    table.add_row(vec!["", "", ""]);
}
table
    .add_row(vec!["", "", ""])
    .add_row(vec!["", "", ""]);
```

With the changes introduced by this PR, we could write the above code like:

```rust
let mut table = Table::new();
table
    .set_header(&vec!["Header1", "Header2", "Header3"])
    .add_row(&vec![
        "This is a text",
        "This is another text",
        "This is the third text",
    ]);
    .add_row_if(|| predicate, vec!["", "", ""])
    .add_row(vec!["", "", ""])
    .add_row(vec!["", "", ""]);
```

The same feature is implemented for `add_rows_if`. I also added some tests, which should cover the newly added feature.

### CODE REVIEW

**Comment by Techassi** [comment]
Is there anything blocking this PR? Is there anything I can do to push this forward @Nukesor?

**Comment by Nukesor** [comment]
I think this could be neat, but I don't like that the fact that the predicate doesn't take any parameters at all.
Right now, you could just accept a boolean instead of a closure and pre-compute the boolean beforehand.

What do you think about the following to make this a bit more useful:
The predicate could get the row/rows as an argument. The user could then decide whether to ignore or use that info.

**Comment by Techassi** [comment]
Yeah I was thinking the same thing, the predicate doesn't do much at the moment. I was also thinking about passing in some kind of "context". I like the idea of passing in row information! I will look into it and will update this PR accordingly.

**Comment by Techassi** [comment]
I updated the closure function signature from

```rust
Fn() -> bool
```

to

```rust
Fn(usize, &vec<Row>) -> bool
```

User now have access the the current row index and all previously added rows. Let me know, if these parameters make sense.

**Comment by Techassi** [comment]
Can we move forward with these changes @Nukesor?

File: src/table.rs
Comment (Techassi) [reply src/table.rs:207]: Yeah you are right. Providing the currently to be added row as the context seems like the better idea. The index should stay in my opinion. Having access to it seems like a good idea. I will adjust the PR as suggested.

Comment (Techassi) [reply src/table.rs]: I added the suggested changes in dca95d9. This should now be resolved.

**Comment by Techassi** [comment]
Let me know if there is anything left to add to this PR @Nukesor before merging!

**Review by Nukesor — APPROVED** [review]
Nice :)

I think I'm happy with the current state. Thanks for your contribution and sorry for taking so long :sweat_smile:

**Comment by Techassi** [comment]
No worries! Love to hear that!

**Pull request merged**
